src/Fixer/Whitespace/MethodChainingIndentationFixer.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Fixer for rules defined in PSR2 ¶2.4.
+ *
+ * @author Vladimir Boliev <voff.web@gmail.com>
+ */
+final class MethodChainingIndentationFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Method chaining MUST be indented with one tab.',
+            array(new CodeSample("<?php\n\$user->setEmail('voff.web@gmail.com')\n         ->setPassword('233434');"),
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // fixer is idempotent
+        return 0;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_OBJECT_OPERATOR);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $indent = $this->whitespacesConfig->getIndent();
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index = 1; $index < count($tokens); ++$index) {
+            if ($tokens[$index]->equals(array(T_OBJECT_OPERATOR))) {
+                if ($this->needLineBreak($index - 1, $tokens)) {
+                    $tokens[$index - 1]->setContent($tokens[$index - 1]->getContent().$lineEnding);
+                    --$index;
+                    continue;
+                }
+
+                $prev = $tokens[$index - 1];
+                $currentWhitespaces = $this->isLineBreak($prev);
+
+                if ($currentWhitespaces !== false) {
+                    $prevMeaningIndex = $tokens->getPrevMeaningfulToken($index);
+                    $rightWhitespaces = $this->getRightIndents($prevMeaningIndex, $tokens, $indent);
+
+                    if ($currentWhitespaces !== $rightWhitespaces) {
+                        $prev->setContent($lineEnding.$rightWhitespaces);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     * @param mixed  $indent
```

Comment:
Expected type is `string`. But I would remove the parameter as it can be retrieved from `$this->whitespacesConfig`.